internal/restconfig/restconfig.go

```diff
@@ -542,35 +538,19 @@ func clusterNameFromContext(rawConfig *api.Config, overridesContext string) *str
 	return &configContext.Cluster
 }
 
-func (rcp *Producer) CheckVersionMismatch(cmd *cobra.Command, args []string) error {
-	return rcp.RunOnSelectedContext(func(clusterInfo *cluster.Info, namespace string, status reporter.Interface) error {
-		crClient := clusterInfo.ClientProducer.ForGeneral()
-
-		submariner := &v1alpha1.Submariner{}
-		err := crClient.Get(context.TODO(), controllerClient.ObjectKey{
-			Namespace: constants.OperatorNamespace,
-			Name:      names.SubmarinerCrName,
-		}, submariner)
-
-		if apierrors.IsNotFound(err) || meta.IsNoMatchError(err) {
-			return nil
-		}
-
-		exit.OnErrorWithMessage(err, fmt.Sprintf("Error retrieving Submariner object %s", names.SubmarinerCrName))
-
-		if submariner != nil && submariner.Spec.Version != "" {
-			subctlVer, _ := semver.NewVersion(version.Version)
-			submarinerVer, _ := semver.NewVersion(submariner.Spec.Version)
+func checkVersionMismatch(submariner *v1alpha1.Submariner) error {
+	if submariner.Spec.Version != "" {
```

Comment:
I'd check if the passed argument isn't empty on the outside, though looking at `semver.NewVersion` this isn't necessary since an empty version (if that even makes sense) will result in an error just as any other malformed version passed to it, and we just skip the check in that case, so you can just skip this check altogether.